src/schema/v2/sales.ts

```diff
@@ -31,38 +31,30 @@ export const SalesConnectionField: GraphQLFieldConfig<void, ResolverContext> = {
     isAuction: {
       description: "Limit by auction.",
       type: GraphQLBoolean,
+      defaultValue: true,
     },
     live: {
       description: "Limit by live status.",
       type: GraphQLBoolean,
+      defaultValue: true,
     },
     published: {
       description: "Limit by published status.",
       type: GraphQLBoolean,
+      defaultValue: true,
     },
     sort: SaleSorts,
   }),
   resolve: async (
     _root,
     { ids, isAuction, live, published, sort, ...paginationArgs },
-    {
-      userRoles,
-      unauthenticatedLoaders: { salesLoaderWithHeaders: loaderUnauthenticated },
-      authenticatedLoaders: { salesLoaderWithHeaders: loaderAuthenticated },
-    }
+    { salesLoaderWithHeaders }
   ) => {
-    paginationArgs.first = paginationArgs.first || 20
     const { page, size, offset } = convertConnectionArgsToGravityArgs(
       paginationArgs
     )
-    let loader
-    if (userRoles.includes("team") && ids.length) {
```

Comment:
In local testing, this was showing up as `undefined` for requests with an admin access token.

Reply:
`userRoles` was undefined? That _should_ have a default of `[]` (from https://github.com/artsy/metaphysics/pull/2518/files#diff-f41e9d04a45c83f3b6f6e630f10117feL214 )

Might it be the `ids` arg which is undefined when not specified? (and so `ids.length` would throw).

Reply:
Hmm, that's possible. I put a `console.log` in to see what `userRoles` was, and it _was_ `undefined`. Not sure! Happy to dig into this later.

Reply:
There might be some seemingly intermittent behavior since when a logged in user _lacks_ the `team` role, this boolean will short circuit and so the problematic `ids.length` wouldn't be evaluated (and so things would work).

But, any logged-in user with a `team` role, and _not_ specifying `ids`, would probably crash.

(this is just from my 👁️ re-reading the code).